// @vonage/voice/dist/cjs/ncco.cjs
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Talk = exports.Stream = exports.Record = exports.Notify = exports.NCCOBuilder = exports.Input = exports.Conversation = exports.Connect = void 0;
const debug_1 = __importDefault(require("debug"));
(0, debug_1.default)('@vonage/voice')('This file is deprecated please update to use the core package');
var Connect_js_1 = require("./classes/NCCO/Connect.js");
Object.defineProperty(exports, "Connect", { enumerable: true, get: function () { return Connect_js_1.Connect; } });
var Conversation_js_1 = require("./classes/NCCO/Conversation.js");
Object.defineProperty(exports, "Conversation", { enumerable: true, get: function () { return Conversation_js_1.Conversation; } });
var Input_js_1 = require("./classes/NCCO/Input.js");
Object.defineProperty(exports, "Input", { enumerable: true, get: function () { return Input_js_1.Input; } });
var NCCOBuilder_js_1 = require("./classes/NCCO/NCCOBuilder.js");
Object.defineProperty(exports, "NCCOBuilder", { enumerable: true, get: function () { return NCCOBuilder_js_1.NCCOBuilder; } });
var Notify_js_1 = require("./classes/NCCO/Notify.js");
Object.defineProperty(exports, "Notify", { enumerable: true, get: function () { return Notify_js_1.Notify; } });
var Record_js_1 = require("./classes/NCCO/Record.js");
Object.defineProperty(exports, "Record", { enumerable: true, get: function () { return Record_js_1.Record; } });
var Stream_js_1 = require("./classes/NCCO/Stream.js");
Object.defineProperty(exports, "Stream", { enumerable: true, get: function () { return Stream_js_1.Stream; } });
var Talk_js_1 = require("./classes/NCCO/Talk.js");
Object.defineProperty(exports, "Talk", { enumerable: true, get: function () { return Talk_js_1.Talk; } });
